File: app/src/main/java/mishka/rssreader/ui/post/PostViewModelFactory.java
package mishka.rssreader.ui.post;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import mishka.rssreader.data.DataManager;

public class PostViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private int postId;
    private DataManager dataManager;

    public PostViewModelFactory(DataManager dataManager, int postId) {
        this.postId = postId;
        this.dataManager = dataManager;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new PostViewModel(dataManager, postId);
    }
}

File: app/src/main/java/mishka/rssreader/ui/post/PostViewModelFactory.kt
package mishka.rssreader.ui.post;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;
import android.view.View

import mishka.rssreader.data.DataManager;

class PostViewModelFactory(val dataManager: DataManager, val postId: Int) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return PostViewModel(dataManager, postId) as T
    }
}
